skxray/fitting/xrf_model.py

```diff
@@ -1117,12 +1116,12 @@ def linear_spectrum_fitting(spectrum, params, elemental_lines=None,
         if np.sum(total_y[:, i]) == 0:
```

Comment:
You should use `for i, func in enumerate(total_list):` which will reduce the amount of slicing you have to do below.  With clever application of zip could probably get rid of the slicing all together.